db/sqlglue.c

```diff
@@ -11900,3 +11900,43 @@ void *comdb2_get_ddl_context(char *name)
     }
     return ctx;
 }
+
+int comdb2_check_vtab_access(sqlite3 *db, sqlite3_module *module)
+{
+    HashElem *current;
+
+    if (!gbl_uses_password) {
+        return 0;
+    }
+
+    for (current = sqliteHashFirst(&db->aModule); current;
+         current = sqliteHashNext(current)) {
+        struct Module *mod = sqliteHashData(current);
+        if (module == mod->pModule) {
+            struct sql_thread *thd = pthread_getspecific(query_info_key);
+            int bdberr;
+            int rc;
+
+            if (module->access_flag == CDB2_ALLOW_ALL) {
```

Comment:
do we have CDB2_ALLOW_ALL for any virtual table?

Reply:
No. Any table for which we want no restriction (comdb2_limits table, for instance) would not have its access_flag set (== CDB2_ALLOW_ALL).